internal/service/ec2/vpc.go

```diff
@@ -402,6 +411,10 @@ func resourceVPCRead(d *schema.ResourceData, meta interface{}) error {
 			d.Set("ipv6_cidr_block", a.Ipv6CidrBlock)
 		}
 	}
+	// check for an assigned ipv6_cidr_block for ipv6_cidr_block_network_border_group
+	if v := d.Get("ipv6_cidr_block"); v != "" {
```

Comment:
i believe this should be `assign_generated_ipv6_cidr_block`

Reply:
Agreed, I've changed this now. Many thanks